src/index.js

```diff
@@ -0,0 +1,194 @@
+      // early to prevent any unhandled errors being thrown.
+      if (typeof document === 'undefined' || !document.createElement){
+        return;
+      }
+
+      const instance = this.getInstance();
+      let clickOutsideHandler;
+
+      if(config && typeof config.handleClickOutside === 'function') {
+        clickOutsideHandler = config.handleClickOutside(instance);
+        if(typeof clickOutsideHandler !== 'function') {
+          throw new Error('WrappedComponent lacks a function for processing outside click events specified by the handleClickOutside config option.');
+        }
+      } else if(typeof instance.handleClickOutside === 'function') {
+        if (Component.prototype.isPrototypeOf(instance)) {
+          clickOutsideHandler = instance.handleClickOutside.bind(instance);
+        } else {
+          clickOutsideHandler = instance.handleClickOutside;
+        }
+      } else if(typeof instance.props.handleClickOutside === 'function') {
+        clickOutsideHandler = instance.props.handleClickOutside;
+      } else {
+        throw new Error('WrappedComponent lacks a handleClickOutside(event) function for processing outside click events.');
+      }
+
+      // TODO: try to get rid of this, could be done with function ref, might be problematic for SFC though, they do not expose refs
+      const componentNode = this.instanceRef;
+      if (componentNode === null) {
+        console.warn('Antipattern warning: there was no DOM node associated with the component that is being wrapped by outsideClick.');
+        console.warn([
+          'This is typically caused by having a component that starts life with a render function that',
+          'returns `null` (due to a state or props value), so that the component \'exist\' in the React',
+          'chain of components, but not in the DOM.\n\nInstead, you need to refactor your code so that the',
+          'decision of whether or not to show your component is handled by the parent, in their render()',
+          'function.\n\nIn code, rather than:\n\n  A{render(){return check? <.../> : null;}\n  B{render(){<A check=... />}\n\nmake sure that you',
+          'use:\n\n  A{render(){return <.../>}\n  B{render(){return <...>{ check ? <A/> : null }<...>}}\n\nThat is:',
+          'the parent is always responsible for deciding whether or not to render any of its children.',
+          'It is not the child\'s responsibility to decide whether a render instruction from above should',
+          'get ignored or not by returning `null`.\n\nWhen any component gets its render() function called,',
+          'that is the signal that it should be rendering its part of the UI. It may in turn decide not to',
+          'render all of *its* children, but it should never return `null` for itself. It is not responsible',
+          'for that decision.'
+        ].join(' '));
+      }
+
+      const fn = this.__outsideClickHandler = generateOutsideCheck(
```

Comment:
such code is on the `master` and didnt change it in any way, although it looked a little weird to me, thought it could have been structured this way to save characters later